plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -284,7 +285,7 @@ defaultSlippage :: Slippage
 defaultSlippage = 200
 
 -- | The CEK machine is parameterized over an emitter function, similar to 'CekBudgetSpender'.
-type CekEmitter uni fun s = Text -> CekM uni fun s ()
+type CekEmitter uni fun s = DList Text -> CekM uni fun s ()
```

Comment:
It's a bit sad that the monomorphization of `readKnown` means we have to commit to `Writer (DList Text)` and then that leaks up here too. But it would be annoying to traverse the list of logs to emit them in the machine's emitter. That said, that *happens* anyway in e.g. some of the log emitters for the CEK machine that assume they can post-process each log line individually. So maybe it's a false saving and it would be cleaner to just traverse the `DList` of logs and emit them again (given that most such lists should be small).

Reply:
Thinking about it, I feel like it was actually stupid to add `DList` here. A call to `traverse_` has to be faster than a call to `?cekEmitter` and in the vast majority of cases we're going to end up with no logs returned from the builtin application machinery. I'll remove that `DList`.

Reply:
Also in the CK machine for consistency?

Reply:
Done.

Reply:
Can you add a comment here regardless of what you do?